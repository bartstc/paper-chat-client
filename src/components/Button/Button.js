import React from 'react';
import PropTypes from 'prop-types';
import { ButtonWrapper } from './Button.styles';

const Button = ({ disabled, onClick, btnType, type, children }) => (
  <ButtonWrapper
    disabled={disabled}
    onClick={onClick}
    className={[btnType].join(' ')}
    type={type}
  >
    {children}
  </ButtonWrapper>
);

Button.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  btnType: PropTypes.string,
  type: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

Button.defaultProps = {
  type: 'button',
};

export default Button;
