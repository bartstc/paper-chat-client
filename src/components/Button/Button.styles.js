import styled from 'styled-components';
import { color } from '../../utils/styles';

export const ButtonWrapper = styled.button`
  width: 100%;
  border: 2px solid ${color.accent};
  background: ${color.accent};
  color: ${color.white};
  padding: 0.8em 1.9em;
  margin-top: 2em;
  text-transform: uppercase;
  font-weight: 700;
  font-size: 0.9rem;
  cursor: pointer;
  transition: all 0.15s ease-in-out;

  &:hover {
    background: ${color.white};
    color: ${color.accent};
  }
`;
