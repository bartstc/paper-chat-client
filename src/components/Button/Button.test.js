import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from './Button';

describe('<Button />', () => {
  const props = {
    btnType: 'btn'
  };

  let wrapper = mount(<Button {...props} />);

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('button contains `btn` class', () => {
    expect(wrapper.find('button').hasClass('btn')).toEqual(true);
  });

  it('has `submit` type', () => {
    expect(wrapper.find('button').type()).toEqual('button');
  });

  describe('when click', () => {
    const mockClick = jest.fn();

    beforeEach(() => {
      const props = { onClick: mockClick };
      wrapper = mount(<Button {...props} />);

      wrapper.find('button').simulate('click');
    });

    it('runs passed function', () => {
      expect(mockClick).toHaveBeenCalled();
    });
  });
});
