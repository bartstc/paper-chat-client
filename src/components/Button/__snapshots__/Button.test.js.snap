// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<Button
  btnType="btn"
  type="button"
>
  <styled.button
    className="btn"
    type="button"
  >
    <StyledComponent
      className="btn"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "bCIpgJ",
            "rules": Array [
              "
  width: 100%;
  border: 2px solid ",
              "#0075C4",
              ";
  background: ",
              "#0075C4",
              ";
  color: ",
              "#FFF",
              ";
  padding: 0.8em 1.9em;
  margin-top: 2em;
  text-transform: uppercase;
  font-weight: 700;
  font-size: 0.9rem;
  cursor: pointer;
  transition: all 0.15s ease-in-out;

  &:hover {
    background: ",
              "#FFF",
              ";
    color: ",
              "#0075C4",
              ";
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="button"
    >
      <button
        className="btn sc-bdVaJa bCIpgJ"
        type="button"
      />
    </StyledComponent>
  </styled.button>
</Button>
`;
